### ISSUE TITLE
[Bug]: The resource aws_instance does not import network interface block.

### ISSUE THREAD

**Issue by rutujachaudhari124** [hashicorp/terraform-provider-aws#30055]
### Terraform Core Version

0.12.31

### AWS Provider Version

2.70.0

### Affected Resource(s)

aws_instance

### Expected Behavior

When resource aws_instance is imported, the network interface block must be added to the state file with other parameters which contains the information of the network interface attached, similar to the ebs block devices.

**Terraform import should add details in such format in the state file after import**:

"network_interface": [
              {
                "delete_on_termination": false,
                "device_index": 0,
                "network_interface_id": "*********************"
              }
            ],

### Actual Behavior

When this resource aws_instance is imported, in the tf state file we can see that the network_interface details are not added. It is mentioned as **network_interface = []**.

The id of primary network_interface is visible as an attribute in the state file, but the network interface block seems empty.

### Relevant Error/Panic Output Snippet

```shell
When we add the network interface configuration, and perform terraform apply, the force replacement is observed and we can see 1 to add and 1 to destroy. It tries to destroy and recreate.

Error:
 + network_interface { # forces replacement
          + delete_on_termination = false
          + device_index          = 0
          + network_interface_id  = "****************"
        }

      ~ root_block_device {
          ~ delete_on_termination = true -> (known after apply)
          ~ device_name           = "/dev/xvda" -> (known after apply)
          ~ encrypted             = false -> (known after apply)
          ~ iops                  = 100 -> (known after apply)
          + kms_key_id            = (known after apply)
          ~ volume_id             = "***************" -> (known after apply)
          ~ volume_size           = 8 -> (known after apply)
          ~ volume_type           = "gp2" -> (known after apply)
        }

      - timeouts {}
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

### Terraform Configuration Files

```terraform
resource "aws_instance" "instance-01" {
  ami           = "*******************"
  instance_type = "t2.micro"
  
  network_interface {
    network_interface_id = "******************"
    device_index         = 0
  }

  tags = {
    Name = "instance-01"
  }
}
```

### Steps to Reproduce

Step 1: create instance 
Step 2: import instance using command : **terraform import aws_instance.instance-01 <instance-id>**
Step 3: terraform apply

### References

https://github.com/hashicorp/terraform-provider-aws/issues/1446

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @rutujachaudhari124 👋 Thank you for taking the time to raise this! It looks like you're using a fairly old version of both Terraform core and the AWS Provider. Are you able to test on a more recent version to verify if the import is successful?

**Comment by rutujachaudhari124** [comment]
yes, I have checked in the latest version (4.60.0), still could not resolve this issue. The network interface is still not added to state file.

**Comment by danilobovo** [comment]
I'm using the latest version of terraform provider (v5.64.0) and the issue still occurring.

**Comment by jscherrer1** [comment]
Hello. I am using v6.12.0 of the Terraform AWS provider and this issue is still occurring.  When we attempt to import an instance into our state, we are still getting an issue with terraform wanting to force replace the entire resource, since the network_interface block is empty. However, we are noticing that the primary_network_interface block is not empty after an import.
